gnucash_uk_vat/config.py

```diff
@@ -135,7 +135,8 @@ def initialise_config(config_path: Path, user: Optional['Config']) -> None:
             "user": getpass.getuser(),
             "local-ip": local_ip,
             "mac-address": mac,
-            "time": now().isoformat()[:-3] + "Z"
+            # https://developer.service.hmrc.gov.uk/guides/fraud-prevention/connection-method/other-direct/#gov-client-local-ips-timestamp
+            "time": now().strftime("%Y-%m-%dT%H:%M:%S.%fZ")
```

Comment:
This creates a string with microsecond precision in the timestamp...
```
>>> now = lambda: datetime.datetime.now()
>>> now().strftime("%Y-%m-%dT%H:%M:%S.%fZ")
'2025-08-06T09:28:51.898608Z'
```

The spec asks for millisecond: `yyyy-MM-ddThh:mm:ss.sssZ`

https://developer.service.hmrc.gov.uk/guides/fraud-prevention/connection-method/desktop-app-direct/

Reply:
That's still using isoformat though, which explicitly fails when tested against the fraud headers API. This format does not report any errors when tested against the API.

Reply:
I suspect their actual check accepts any number of digits after the decimal point.

Reply:
I just that problem too, it's fixed in master

Reply:
> I made an agreement saying that the spec would be followed by this project

Yes, my concern is that it currently deviates completely from the spec and explicitly fails validation. My only concern is that adding string manipulation in may cause the same problem again in the future.

If you want to insist on millisecond format despite their verification allowing microseconds, then we can add the string manipulation back in. I'm just voicing my concern that this practice has already caused a deviation and complete failure of the format in the past..

I'll leave the fix here as a suggestion. If you still want to go with millisecond then just commit that and merge:

```suggestion
            "time": now().strftime("%Y-%m-%dT%H:%M:%S.%f")[:-3] + "Z"
```